pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -876,3 +879,104 @@ func (c updateRulesOrderInGroup) Exec(sess *xorm.Session, migrator *migrator.Mig
 	}
 	return nil
 }
+
+func AddExternalAlertmanagerToDatasourceMigration(mg *migrator.Migrator) {
+	mg.AddMigration("migrate external alertmanagers to datsourcse", &externalAlertmanagerToDatasources{})
+}
+
+type externalAlertmanagerToDatasources struct {
+	migrator.MigrationBase
+}
+
+// Just copy old struct code to use Alertmanagers as new one is not getting it.
+type AdminConfiguration struct {
+	ID    int64 `xorm:"pk autoincr 'id'"`
+	OrgID int64 `xorm:"org_id"`
+
+	Alertmanagers []string
+
+	// SendAlertsTo indicates which set of alertmanagers will handle the alert.
+	SendAlertsTo ngmodels.AlertmanagersChoice `xorm:"send_alerts_to"`
+
+	CreatedAt int64 `xorm:"created"`
+	UpdatedAt int64 `xorm:"updated"`
+}
+
+func (e externalAlertmanagerToDatasources) SQL(dialect migrator.Dialect) string {
+	return "migrate external alertmanagers to datasource"
+}
+
+func (e externalAlertmanagerToDatasources) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	var results []AdminConfiguration
+	err := sess.SQL("SELECT org_id, alertmanagers FROM ngalert_configuration").Find(&results)
+	if err != nil {
+		return err
+	}
+
+	for _, result := range results {
+		for _, am := range result.Alertmanagers {
+			u, err := url.Parse(am)
+			if err != nil {
+				return err
+			}
+			uri := fmt.Sprintf("%s://%s", u.Scheme, u.Host)
```

Comment:
What about the port and path?

Reply:
`u.Host` already includes the port but you are right about the path. Someone could have a reverse proxy pointing to an AM url.